src/gui/panels/base_panel.py

```diff
@@ -227,3 +227,19 @@ def add_text_label(
             flags=wx.SizerFlags(0).Border(
                 direction=(wx.TOP | wx.BOTTOM),
                 borderinpixels=BasePanel.DEFAULT_SPACING_PX_LINE_TOP_BOTTOM).Expand())
+
+    @staticmethod
+    def add_toggle_button(
+        parent: wx.Window,
+        sizer: wx.BoxSizer,
+        label: str
+    ) -> wx.ToggleButton:
```

Comment:
This looks useful, probably better than some of the checkboxes which I've used elsewhere. Thanks!